Directory.Build.props

```diff
@@ -151,11 +151,14 @@
 
   <PropertyGroup Label="CalculateOS">
     <!-- Default to portable build if not explicitly set -->
+    <PortableBuild Condition="'$(PortableBuild)' == '' and '$(DotNetBuildFromSource)' == 'true'">false</PortableBuild>
     <PortableBuild Condition="'$(PortableBuild)' == ''">true</PortableBuild>
 
+    <_hostRid Condition="'$(MSBuildRuntimeType)' == 'core'">$([System.Runtime.InteropServices.RuntimeInformation]::RuntimeIdentifier)</_hostRid>
```

Comment:
I think we can improve how we're processing rids in this file, but I'm not going to make a brave attempt as part of this PR.
I've limited myself to adding cases guarded by `'$(DotNetBuildFromSource)' == 'true'` and use appropriate defaults for properties source-build sets.